camel-core/src/test/java/org/apache/camel/component/file/FileConsumePollEnrichFilePredicateTest.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.file;
+
+import java.util.function.Predicate;
+import org.apache.camel.ContextTestSupport;
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+
+public class FileConsumePollEnrichFilePredicateTest extends ContextTestSupport {
+
+    @Override
+    protected void setUp() throws Exception {
+        deleteDirectory("target/enrich");
+        deleteDirectory("target/enrichdata");
+        super.setUp();
+    }
+
+    public void testPollEnrichFileReject() throws Exception {
+        MockEndpoint mock = getMockEndpoint("mock:result");
+        mock.expectedMessageCount(1);
+        mock.expectedBodiesReceived((Object) null);
+
+        template.sendBody("seda:start", "Start");
+        log.info("Sleeping for 1/4 sec before writing enrichdata file");
+        Thread.sleep(250);
+        template.sendBodyAndHeader("file://target/enrichdata", "O", Exchange.FILE_NAME, "AAA.dat");
+        log.info("... write done");
+
+        assertMockEndpointsSatisfied();
+    }
+
+    public void testPollEnrichFileAccept() throws Exception {
+        MockEndpoint mock = getMockEndpoint("mock:result");
+        mock.expectedMessageCount(1);
+        mock.expectedBodiesReceived("Hi Camel!");
+
+        template.sendBody("seda:start", "Start");
+        log.info("Sleeping for 1/4 sec before writing enrichdata file");
+        Thread.sleep(250);
+        template.sendBodyAndHeader("file://target/enrichdata", "Hi Camel!", Exchange.FILE_NAME, "AAA.dat");
+        log.info("... write done");
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Override
+    protected RouteBuilder createRouteBuilder() throws Exception {
+        return new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                from("seda:start")
+                    .setHeader(Exchange.FILE_PREDICATE, () -> (Predicate<GenericFile>) genericFile -> genericFile.getFileLength() > 4)
```

Comment:
No we should not use a complex object like a predicate in a header, its for smaller objects like a text etc.

Reply:
I understand your point, but as the static typing did not prevent me, I kind of abused it :)

What do you think about implementing this in the DSL (PollEnrichDefinition), so its possible to define a Predicate for filtering the exchanges handled by the PollEnricher before even aggregating into the original exchange?

Reply:
No I dont think we should complicate things even more, lets close this PR and the JIRA ticket at this moment.